boilerplate/index.ts

```diff
@@ -564,49 +575,51 @@ export function createBoilerplate(sdk: CDP) {
 
                         const allCloudStorageRemoteApplications = await bOps.applications.getAll().then(apps => apps.filter(app => app.type === 'CloudStorage'));
 
-
                         const remoteConnectors = await sdk.api.workspaces.for(config.workspaceId).applibrary.getAll({includePublic: true});
 
-                        const boilerplateConnectorTypes: CSType[] = [ 'AWS S3', 'Microsoft Azure Blob', 'Google Cloud Storage', 'SFTP']
+                        const boilerplateConnectorTypes: CSType[] = ['AWS S3', 'Microsoft Azure Blob', 'Google Cloud Storage', 'SFTP']
 
                         // get remote connectors that are Cloud Storage connectors
-                        const remoteCloudStorageConnectors = remoteConnectors &&
-                            (remoteConnectors?.filter(connector => connector.type === 'CloudStorage' && (boilerplateConnectorTypes.includes(connector.name as CSType))));
-
-                        remoteCloudStorageConnectors.map(async connector => {
+                        const remoteCloudStorageConnectors =
+                            keepUnique(remoteConnectors, conn => conn.name).filter((connector: Connector) => connector.type === 'CloudStorage' && connector.enabled && (boilerplateConnectorTypes.includes(connector.name as CSType)));
+                        await Promise.all(remoteCloudStorageConnectors.map(async (connector: Connector) => {
                             // get the corresponding cloud storage application
 
-                            //get cloud storage applications with the same type as connector // npt working
-                            let remoteCloudStorageApplication = allCloudStorageRemoteApplications?.find(application => (application['originConnectorId'] === connector.id))
-
-                            let remoteCloudStorageApplicationId: ApplicationId = remoteCloudStorageApplication?.id;
+                            // get cloud storage applications created from connector
+                            let remoteCloudStorageApplication = allCloudStorageRemoteApplications?.find(application => (application['originConnectorId'] === connector.id || application.name === connector.name))
```

Comment:
`originConnectorId` added to `Application` definition.
also - i think you can drop the part in the condition regarding the name.

Reply:
regarding originConnectorId --> see slack messages